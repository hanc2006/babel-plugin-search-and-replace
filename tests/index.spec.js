const babel = require("@babel/core");
const searchAndReplaceplugin = require("../lib/index");

var example = `
var foo = 1;
if (foo) console.log(foo);
const hello = "hello";

const myFunc = function(param = "hello"){
    return 1;
}
`;

it("should replace nothing", () => {
  const { code } = babel.transform(example, {
    plugins: [searchAndReplaceplugin]
  });
  expect(code).toMatchSnapshot();
});

it("should replace foo with moo", () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          rules: [
            {
              search: "foo",
              replace: "moo",
              flag: "g"
            }
          ]
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});
