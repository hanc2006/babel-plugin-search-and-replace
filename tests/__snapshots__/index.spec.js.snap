// Jest Snapshot

exports[`should replace nothing`] = `
"var foo = 1;
if (foo) console.log(foo);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
  return 1;
};"
`;

exports[`should replace foo with moo`] = `
"var moo = 1;
if (moo) console.log(foo);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
  return 1;
};"
`;
